src/zip.c

```diff
@@ -1835,7 +1840,9 @@ struct zip_t *zip_stream_openwitherror(const char *stream, size_t size,
   }
   zip->level = (mz_uint)level;
 
-  if ((stream != NULL) && (size > 0) && (mode == 'r')) {
+  // for modes 'd' and 'w', would be better to use mz_zip_reader_init_writer, but there's no clean
+  // way to load the existing stream with that.
+  if ((stream != NULL) && (size > 0) && (mode == 'r' || mode == 'd' || mode == 'w')) {
     if (!mz_zip_reader_init_mem(&(zip->archive), stream, size, 0)) {
       *errnum = ZIP_ERINIT;
```

Comment:
If stream is NULL, it would jump to the else if branch

Reply:
What I was trying to achieve with the logic is:
  if stream is NOT NULL
     load it so that we may read the zip or make updates to the zip (either writes or deletes)
  if stream is NULL
     then a user would only want create a new zip / delete